examples/tunables_limit_memory.rs

```diff
@@ -0,0 +1,131 @@
+use std::sync::Arc;
+use wasmer::{
+    imports, wat2wasm, Instance, Memory, MemoryError, MemoryType, Module, Pages, Store, TableType,
+    Tunables as BaseTunables,
+};
+use wasmer_compiler::Target;
+use wasmer_compiler_cranelift::Cranelift;
+use wasmer_engine::Tunables;
+use wasmer_engine_jit::JIT;
+use wasmer_vm::{Memory as MemoryTrait, MemoryStyle, Table, TableStyle};
```

Comment:
For the example, I think it would be better to get some of the structs directly from Wasmer (so people don't depend on other subpackages).

```suggestion
use wasmer::{
    imports, wat2wasm, Instance, Memory, MemoryError, MemoryType, Module, Pages, Store, TableType,
    Tunables, WasmerTunables, Target, Memory, RuntimeMemory, MemoryStyle, RuntimeTable, TableStyle
};
use wasmer_compiler_cranelift::Cranelift;
use wasmer_engine_jit::JIT;
```

This will imply exposing the following structs in the Wasmer API (`lib/api/src/lib.rs`):

```rust
pub use crate::tunables::Tunables as WasmerTunables;

pub use wasmer_engine::Tunables;
pub use wasmer_vm::{Memory as RunnableMemory, MemoryStyle, Table as RunnableTable, TableStyle};
```

-------

Note: other alternative would be exposing the subcrates as submodules off the API:

```suggestion
use wasmer::{
    imports, wat2wasm, Instance, Memory, MemoryError, MemoryType, Module, Pages, Store, TableType,
    Target, Tunables as BaseTunables,
    engine::Tunables,
    vm::{Memory as MemoryTrait, MemoryStyle, Table, TableStyle};
};
use wasmer_compiler_cranelift::Cranelift;
use wasmer_engine_jit::JIT;
```

```rust
pub mod engine {
    pub use wasmer_engine::Tunables;
}
pub mod vm {
    pub use wasmer_vm::{Memory, MemoryStyle, Table, TableStyle};
}
```

Thoughts @MarkMcCaskey ?

Reply:
Yeah I can see either way, for things that conflict in name like `Memory` namespacing can be a nice way to deal with it, I think expecting people to know about `use x as y` syntax in Rust is probably not a good assumption (though maybe the error message you get tells you how to fix it), so I'd be more in favor of the public submodule idea than exporting 2 things with the same name at the same scope level.

Reply:
> I think expecting people to know about use x as y syntax in Rust is probably not a good assumption (though maybe the error message you get tells you how to fix it)

Agreed. However, in my case the bigger problem was very very confusing error messages about types that are `Sized` or size not known at compile time when I had `Memory` (trait) importet and then used `Vec<Memory>` in the code, which referes to the exported `Memory` (struct). Took a while to realize there are more than one of those in Wasmer.